src/components/general/DimensionalBarGraph.tsx

```diff
@@ -0,0 +1,229 @@
+
+function getDefaultSliceConfig(
+  cube: DimensionalMetric,
+  activeGoal: InstanceGoal | null
+) {
+  /**
+   * By default, we group by the first dimension `metric` has, whatever it is.
+   * @todo Is there a better way to select the default?
+   *
+   * If the currently selected goal has category selections for this metric,
+   * we might choose another dimension.
+   *
+   * NOTE: This is just the default -- the actually active filtering and
+   * grouping is controlled by the `sliceConfig` state below.
+   */
+  const defaultConfig: SliceConfig = {
+    dimensionId: cube.dimensions[0]?.id,
+    categories: {},
+  };
+
+  if (!activeGoal) return defaultConfig;
+
+  const cubeDefault = cube.getChoicesForGoal(activeGoal);
+  if (!cubeDefault) return defaultConfig;
+  defaultConfig.categories = cubeDefault;
+  /**
+   * Check if our default dimension to slice by is affected by the
+   * goal-based default filters. If so, we should choose another
+   * dimension.
+   */
+  if (
+    defaultConfig.dimensionId &&
+    cubeDefault.hasOwnProperty(defaultConfig.dimensionId)
+  ) {
+    const firstPossible = cube.dimensions.find(
+      (dim) => !cubeDefault.hasOwnProperty(dim.id)
+    );
+    defaultConfig.dimensionId = firstPossible?.id;
+  }
+  return defaultConfig;
+}
+
+type DimensionalBarGraphProps = {
+  metric: NonNullable<DimensionalNodeMetricFragment['metricDim']>;
+  endYear: number;
+  color?: string | null;
+};
+
+const DimensionalBarGraph = ({ metric, endYear }: DimensionalBarGraphProps) => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const activeGoal = useReactiveVar(activeGoalVar);
+  const cube = useMemo(() => new DimensionalMetric(metric), [metric]);
+  const defaultConfig = getDefaultSliceConfig(cube, activeGoal);
+  const [sliceConfig, setSliceConfig] = useState<SliceConfig>(defaultConfig);
+
+  useEffect(() => {
+    /**
+     * If the active goal changes, we will reset the grouping + filtering
+     * to be compatible with the new choices (if the new goal has common
+     * dimensions with our metric).
+     */
+    if (!activeGoal) return;
+    const newDefault = getDefaultSliceConfig(cube, activeGoal);
+    if (!newDefault || isEqual(sliceConfig, newDefault)) return;
+    setSliceConfig(newDefault);
+  }, [activeGoal, cube]);
+
+  const yearData = cube.getSingleYear(endYear, sliceConfig.categories);
+
+  const plotData: Partial<Plotly.PlotData>[] = [];
+
+  let longUnit = metric.unit.htmlShort;
+  // FIXME: Nasty hack to show 'CO2e' where it might be applicable until
+  // the backend gets proper support for unit specifiers.
+  if (
+    cube.hasDimension('emission_scope') &&
+    !cube.hasDimension('greenhouse_gases')
+  ) {
+    if (metric.unit.short === 't/Einw./a') {
+      longUnit = t('tco2-e-inhabitant');
+    } else if (metric.unit.short === 'kt/a') {
+      longUnit = t('ktco2-e');
+    }
+  }
+
+  let maxTotal = 0;
+  yearData.categoryTypes[1].options.forEach((colId, cIdx) => {
```

Comment:
Is accessing the `categoryTypes` by index safe for future customers who use this bar chart? Or Zürich specific?

Reply:
This is Zurich specific for now ( we have hard coded if zurich to display the bar chart ). Although as far as I understand the dimensional graph, we are handling a cube here that always should have two dimensions (categories) so array should always be limited to two items...